pkg/lifecycle/service.go

```diff
@@ -46,25 +101,37 @@ type ProcessorService interface {
 	MakeRunnableProcessor(ctx context.Context, i *processor.Instance) (*processor.RunnableProcessor, error)
 }
 
-// PluginDispenserFetcher can fetch a plugin.
-type PluginDispenserFetcher interface {
+// ConnectorPluginService can create a connector plugin dispenser.
+type ConnectorPluginService interface {
 	NewDispenser(logger log.CtxLogger, name string, connectorID string) (connectorPlugin.Dispenser, error)
 }
 
+// PipelineService can fetch, list and update the status of a pipeline instance.
+type PipelineService interface {
+	Get(ctx context.Context, pipelineID string) (*pipeline.Instance, error)
+	List(ctx context.Context) map[string]*pipeline.Instance
+	UpdateStatus(ctx context.Context, pipelineID string, status pipeline.Status, errMsg string) error
+}
+
+// OnFailure registers a handler for a lifecycle.FailureEvent.
+// Only errors which happen after a pipeline has been started
+// are being sent.
+func (s *Service) OnFailure(handler FailureHandler) {
+	s.handlers = append(s.handlers, handler)
+}
+
 // Run runs pipelines that had the running state in store.
 func (s *Service) Run(
```

Comment:
I'd think about renaming this to `Init`. Previously this was on the `pipeline.Service` where we already had an `Init` method, so we had to find a different name. In fact, all services have an `Init` method, which initializes the service (e.g. loads stuff from the store), that's why I think this name could be a better fit.